docs/sources/userguide/plugins.md

```diff
@@ -0,0 +1,157 @@
+Docker Plugins MVP
+==================
+
+Plugins provide a mechanism for hooking into the Docker engine to extend its behavior.
+
+HTTP over a local UNIX socket is used to communicate between Docker and its plugins.
+
+Plugins are distributed as containers.
```

Comment:
I think it's too early to invent a whole new permission system, and a ui
for it, on a short deadline. I don't think we need it for MVP.

We already have an existing permission system: the options to "docker run"
and "docker start". Privileged, net=none, capabilities etc. Since a plugin
is a container, it stands to reason that plugin permissions should simply
be container permissions.



On Thu, Mar 12, 2015 at 3:54 PM, lukemarsden <notifications@github.com>
wrote:

> In docs/sources/userguide/plugins.md
> <https://github.com/cpuguy83/docker/pull/3#discussion_r26352285>:
>
> > @@ -0,0 +1,157 @@
> > +Docker Plugins MVP
> > +==================
> > +
> > +Plugins provide a mechanism for hooking into the Docker engine to extend its behavior.
> > +
> > +HTTP over a local UNIX socket is used to communicate between Docker and its plugins.
> > +
> > +Plugins are distributed as containers.
>
> Hmm. The reason we wanted the Docker socket was so that e.g. extension
> payloads (the messages docker sends to the extensions) could refer to e.g.
> a container ID and the extension would be free to look up the data about
> the container. powerstrip-weave also had a use-case where it was
> necessary to inspect the image JSON for a container as it was being started.
>
> A more sophisticated approach than bind-mounting the docker socket would
> be to define a set of interfaces with different sets of permissions. So for
> this use case we could have a "getContainerJSON" permission. Then we can
> fill this catalogue of interfaces as we come across them.
>
> Or a middle ground could be to:
> a) define permissions, make it explicit that permissions get granted
> somehow, and
> b) define a "can have read/write access to the Docker API" permission.
>
> I'm really keen to keep this MVP so that we can land it in Docker 1.7. I
> suggest we approach it therefore with the above middle-ground: that way
> when users load extensions, they have to explicitly give the plugin
> permission.
>
> Note that plugins will often want to have (effectively, via --privileged)
> root on the host anyway to do system level things.
>
> @progrium <https://github.com/progrium> @shykes
> <https://github.com/shykes> what do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cpuguy83/docker/pull/3/files#r26352285>.
>